src/plots/cartesian/handle_outline.js

```diff
@@ -0,0 +1,95 @@
+/**
+* Copyright 2012-2020, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Registry = require('../../registry');
+
+
+function activateShape(gd, path, drawShapes) {
+    var element = path.node();
+    var id = +element.getAttribute('data-index');
+
+    for(var q = 0; q < gd._fullLayout.shapes.length; q++) {
+        var shapeIn = gd._fullLayout.shapes[q]._input;
+        if(q === id && shapeIn.editable) {
+            gd._fullLayout._activeShapeIndex = q;
+            break;
```

Comment:
Does this need to be a loop? Can't you just inspect `gd._fullLayout.shapes[id]`?

Reply:
Also why do we need to look at `_input.editable`? Does that not make it into `_fullLayout`?

Reply:
Good eye! Revised in https://github.com/plotly/plotly.js/pull/4775/commits/90a7ec7dd1cfc776b880904c7c095a3b342cd0c7.